indexer/services/comlink/src/controllers/api/v4/historical-funding-payment-controller.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  BlockFromDatabase,
+  BlockTable,
+  DEFAULT_POSTGRES_OPTIONS,
+  FundingIndexMap,
+  PerpetualMarketFromDatabase,
+  PerpetualMarketTable,
+  PerpetualPositionFromDatabase,
+  PerpetualPositionStatus,
+  PerpetualPositionTable,
+  SubaccountFromDatabase,
+  SubaccountTable,
+} from '@dydxprotocol-indexer/postgres';
+import express from 'express';
+import { matchedData } from 'express-validator';
+import {
+  Controller, Get, Route, Path, Query,
+} from 'tsoa';
+
+import { getReqRateLimiter } from '../../../caches/rate-limiters';
+import { NotFoundError } from '../../../lib/errors';
+import {
+  getFundingIndexMaps,
+  getPerpetualPositionsWithUpdatedFunding,
+  handleControllerError,
+  initializePerpetualPositionsWithFunding,
+} from '../../../lib/helpers';
+import { rateLimiterMiddleware } from '../../../lib/rate-limit';
+import {
+  CheckSubaccountSchema,
+  CheckTickerParamSchema,
+} from '../../../lib/validation/schemas';
+import { handleValidationErrors } from '../../../request-helpers/error-handler';
+import {
+  HistoricalFundingRequest,
+  HistoricalFundingPaymentResponse,
+  MarketType,
+  PerpetualPositionWithFunding,
+} from '../../../types';
+
+const router: express.Router = express.Router();
+
+@Route('historicalFundingPayment')
+class HistoricalFundingPaymentController extends Controller {
+  @Get('/:ticker')
+  async getHistoricalFundingPayment(
+    @Path() ticker: string,
+      @Query() address: string,
+      @Query() subaccountNumber: number,
+      //   @Query() limit?: number,
+      //   @Query() effectiveBeforeOrAtHeight?: number,
+      //   @Query() effectiveBeforeOrAt?: IsoString,
+  ): Promise<HistoricalFundingPaymentResponse> {
+    const subaccountId: string = SubaccountTable.uuid(
+      address,
+      subaccountNumber,
+    );
+    const [
+      subaccount,
+      perpetualMarket,
+      latestBlock,
+    ] : [
+      SubaccountFromDatabase | undefined,
+      PerpetualMarketFromDatabase | undefined,
+      BlockFromDatabase,
+    ] = await Promise.all([
+      SubaccountTable.findById(subaccountId),
+      PerpetualMarketTable.findByTicker(ticker),
+      BlockTable.getLatest(),
+    ]);
+
+    if (perpetualMarket === undefined) {
+      throw new NotFoundError(
+        `${ticker} not found in markets of type ${MarketType.PERPETUAL}`,
+      );
+    }
+
+    if (subaccount === undefined) {
+      throw new NotFoundError(
+        `No subaccount found with address ${address} and subaccountNumber ${subaccountNumber}`,
+      );
+    }
+
+    // Add tests for pagination, limits & date ranges
```

Comment:
nit: usually we have format TODO(CT-XXX): Add tests... where CT-XXX is a lienar ticket